File: 프로그래머스/0/120890. 가까운 수/가까운 수.py
def solution(array, n):
    answer = 0
    array.sort()
    gap = 100
    for i in array :
        # print(f"gap : {gap}")
        # print(f"answer :{answer}") 
        # print(f"i :{i}")
        if gap == 100 :
            gap = n - i
            answer = i
        
        elif abs(gap) > abs(n - i) :
            gap = n - i
            answer = i                
        
        if i == n :
            answer = i
    
    return answer

File: 프로그래머스/0/120890. 가까운 수/가까운 수.js
function solution(array, n) {
    var answer = 0;
    const minNum = Math.min.apply(null, array.filter(item => item - n >= 0));
    const maxNum = Math.max.apply(null, array.filter(item => item - n <= 0));
    return answer = (minNum - n) >= Math.abs(n - maxNum) ? maxNum : minNum;
}
